### PR TITLE
Page flicker

### PR DESCRIPTION
**Pull request by leohtkam**
Fixed the page flickering problem where a page is rendered before all content is loaded from database.
Now the old page is shown until all content of the new page is ready and is then rendered.

### CODE REVIEW

File: app/assets/javascripts/homepage/conversation.coffee [hunk 1]
```diff
@@ -9,7 +9,7 @@ conversation.directive("onEscape", ->
         )
 )
 
-conversation.controller("ConversationController", ["$scope", "$stateParams", "API", "TimeUtil", "AppState", ($scope, $stateParams, API, TimeUtil, AppState) ->
+conversation.controller("ConversationController", ["$scope", "$stateParams", "API", "TimeUtil", "AppState", "conversationData", ($scope, $stateParams, API, TimeUtil, AppState, conversationData) ->
```
Comment (kuu2) [inline app/assets/javascripts/homepage/conversation.coffee]: The conversation data shouldn't be in camelcase right?

File: app/assets/javascripts/homepage/profile.coffee [hunk 2]
```diff
@@ -1,6 +1,6 @@
 profile = angular.module("Profile",["SharedServices"])
 
-profile.controller("ProfileController", ["$http", "$location", "$scope", "$stateParams", "API", "AppState", ($http, $location, $scope, $stateParams, API, AppState) ->
+profile.controller("ProfileController", ["$http", "$location", "$scope", "$stateParams", "API", "AppState", "profileData", ($http, $location, $scope, $stateParams, API, AppState, profileData) ->
```
Comment (kuu2) [inline app/assets/javascripts/homepage/profile.coffee]: Also fix camelcase here

File: app/assets/javascripts/startpage/surveyQuestions.coffee [hunk 3]
```diff
@@ -13,7 +13,7 @@ surveyQuestions.directive("highlighter", () ->
             )
 )
 
-surveyQuestions.controller("SurveyQuestionsController", ["$scope", "$http", "$state", "$stateParams", "API", "StartPageStaticData", "StartPageStateData", ($scope, $http, $state, $stateParams, API, StartPageStaticData, StartPageStateData) ->
+surveyQuestions.controller("SurveyQuestionsController", ["$scope", "$http", "$state", "$stateParams", "API", "StartPageStaticData", "StartPageStateData", "questionData", ($scope, $http, $state, $stateParams, API, StartPageStaticData, StartPageStateData, questionData) ->
```
Comment (kuu2) [inline app/assets/javascripts/startpage/surveyQuestions.coffee]: Fix camelcase here

File: app/assets/javascripts/startpage/surveyTopics.coffee [hunk 4]
```diff
@@ -1,6 +1,6 @@
 surveyTopics = angular.module("SurveyTopics", ["SharedServices", "StartPageServices"])
 
-surveyTopics.controller("SurveyTopicsController", ["$scope", "$http", "$state", "API", "StartPageStaticData", "StartPageStateData", ($scope, $http, $state, API, StartPageStaticData, StartPageStateData) ->
+surveyTopics.controller("SurveyTopicsController", ["$scope", "$http", "$state", "API", "StartPageStaticData", "StartPageStateData", "topicData", ($scope, $http, $state, API, StartPageStaticData, StartPageStateData, topicData) ->
```
Comment (kuu2) [inline app/assets/javascripts/startpage/surveyTopics.coffee]: Fix camelcase here

File: app/assets/javascripts/homepage/profile.coffee [hunk 5]
```diff
@@ -56,19 +56,22 @@ profile.controller("ProfileController", ["$http", "$location", "$scope", "$state
...
+                parse_profile(profile)
             .error (result, status) ->
                 if result?
                     reason = result.error
                 else
                     reason = "status code #{status}"
                 console.log "profile loading failed: #{reason}"
-    load_profile($scope.profileUID)
+
+    parse_profile = (profile) ->
```
Comment (whsieh) [inline app/assets/javascripts/homepage/profile.coffee]: `parseProfile` as per style guidelines

File: app/assets/javascripts/homepage/conversation.coffee (hunk 1 above)
Comment (whsieh) [inline app/assets/javascripts/homepage/conversation.coffee]: It should be in camelcase.
However, it should also be capitalized.

File: app/assets/javascripts/homepage/conversation.coffee [hunk 6]
```diff
@@ -161,37 +161,40 @@ conversation.controller("ConversationController", ["$scope", "$stateParams", "AP
...
-            lastUpdateTime = TimeUtil.timeSince1970InSeconds()
+            parse_conversation(response, scrollToEnd)
 
     shouldPollConversationData = -> TimeUtil.timeSince1970InSeconds() - lastUpdateTime > CONVERSATION_POLL_PERIOD
     setInterval( ->
         updateConversationData(false) if shouldPollConversationData()
     , CONVERSATION_POLL_PERIOD * 1000)
 
-    updateConversationData(true)
+    parse_conversation = (response, scrollToEnd) ->
```
Comment (whsieh) [inline app/assets/javascripts/homepage/conversation.coffee]: `parseConversation`

**Comment by whsieh** [comment]
awesome +1

**Comment by nickfong** [comment]
+1

**Pull request merged**